File: src-all/Rust/S0959-regions-cut-by-slashes.rs
/*
In a N x N grid composed of 1 x 1 squares, each 1 x 1 square consists of a /, \, or blank space.  These characters divide the square into contiguous regions.

(Note that backslash characters are escaped, so a \ is represented as "\\".)

Return the number of regions.

 

Example 1:
Input:
[
  " /",
  "/ "
]
Output: 2
Explanation: The 2x2 grid is as follows:

Example 2:
Input:
[
  " /",
  "  "
]
Output: 1
Explanation: The 2x2 grid is as follows:

Example 3:
Input:
[
  "\\/",
  "/\\"
]
Output: 4
Explanation: (Recall that because \ characters are escaped, "\\/" refers to \/, and "/\\" refers to /\.)
The 2x2 grid is as follows:

Example 4:
Input:
[
  "/\\",
  "\\/"
]
Output: 5
Explanation: (Recall that because \ characters are escaped, "/\\" refers to /\, and "\\/" refers to \/.)
The 2x2 grid is as follows:

Example 5:
Input:
[
  "//",
  "/ "
]
Output: 3
Explanation: The 2x2 grid is as follows:

 

Note:
	1 <= grid.length == grid[0].length <= 30
	grid[i][j] is either &#39;/&#39;, &#39;\&#39;, or &#39; &#39;.


*/
pub struct Solution {}
impl Solution {
    pub fn regions_by_slashes(grid: Vec<String>) -> i32 {
        
    }
}

fn main() {
  assert_eq!(0, Solution::regions_by_slashes(0));
  println!("Pass test cases!");
}

File: src-all/C++/S0959-regions-cut-by-slashes.cpp
/*
An n x n grid is composed of 1 x 1 squares where each 1 x 1 square consists of a '/', '\', or blank space ' '. These characters divide the square into contiguous regions.

Given the grid grid represented as a string array, return the number of regions.

Note that backslash characters are escaped, so a '\' is represented as '\\'.

 
Example 1:
Input: grid = [" /","/ "]
Output: 2

Example 2:
Input: grid = [" /","  "]
Output: 1

Example 3:
Input: grid = ["/\\","\\/"]
Output: 5
Explanation: Recall that because \ characters are escaped, "\\/" refers to \/, and "/\\" refers to /\.

 
Constraints:
	n == grid.length == grid[i].length
	1 <= n <= 30
	grid[i][j] is either '/', '\', or ' '.


*/
#include <cassert>
#include <iostream>

class Solution {
public:
    int regionsBySlashes(vector<string>& grid) {
        
    }
};

int main() {
  Solution sol;
  assert(sol.regionsBySlashes(0) == 0);
  std::cout << "Pass test cases!" << std::endl;
}
